dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/filter-fee-record-payment-entity-groups.ts

```diff
@@ -1,52 +1,86 @@
-import { PaymentDetailsFilters, PremiumPaymentsFilters } from '@ukef/dtfs2-common';
+import { Currency, PremiumPaymentsFilters } from '@ukef/dtfs2-common';
 import { FeeRecordPaymentEntityGroup } from '../../../../../types/fee-record-payment-entity-group';
+import { ValidatedPaymentDetailsFilters } from '../../../../../types/utilisation-reports';
 
 /**
- * Filters the supplied list of fee record payment entity groups by the {@link PremiumPaymentsFilters}.
- * The facility id filter allows for partial matches.
- * @param feeRecordPaymentEntityGroups - A list of fee record payment entity groups to be filtered
- * @param filters - The filters to be applied to the fee record payment data
- * @param filters.facilityId - The facility ID filter
- * @returns The filtered fee record payment entity groups. If no filters are provided, returns the original list.
+ * The type intersection of the PremiumPaymentsFilters and ValidatedPaymentDetailsFilters.
```

Comment:
I don't think we should describe the implementation here rather explain why the type exists, that way we could add more to the union in the future without having to update the docstring, e.g.
```suggestion
 * All possible filters for FeeRecordPaymentGroup filtering
```